app/managers/performance_metrics_manager/index.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getTimeSinceStartup} from 'react-native-startup-time';
+
+import Batcher from './performance_metrics_batcher';
+
+type Target = 'HOME' | 'CHANNEL' | 'THREAD' | undefined;
+type MetricName = 'mobile_channel_switch' |
+    'mobile_team_switch';
+
+class PerformanceMetricsManager {
+    private startTimes: {[metricName: string]: undefined | number} = {};
+    private target: Target;
+    private batchers: {[serverUrl: string]: Batcher} = {};
+
+    private ensureBatcher(serverUrl: string) {
+        if (this.batchers[serverUrl]) {
+            return this.batchers[serverUrl];
+        }
+
+        this.batchers[serverUrl] = new Batcher(serverUrl);
+        return this.batchers[serverUrl];
+    }
+
+    public setLoadTarget(target: Target) {
+        this.target = target;
+    }
+
+    public finishLoad(location: Target, serverUrl: string) {
+        if (this.target !== location) {
+            return;
+        }
+
+        getTimeSinceStartup().then((measure) => {
+            this.ensureBatcher(serverUrl).addToBatch({
+                metric: 'mobile_load',
+                value: measure / 1000,
```

Comment:
I think it's better to handle dividing these by 1000 on the server to get them in seconds because that means it'll always happen in one place as opposed to needing to divide them in multiple places in both web/mobile apps and in the tests for them.

We use milliseconds for timestamps and durations all over MM, so I think we should be trying to convert these to seconds as close to the Prometheus code as possible